### PR TITLE
Validation and test for stormwater - noninteger soils data

### LINKED ISSUE #1084: Urban Stormwater: soil raster as float type causes index error
A user on the [forum](https://community.naturalcapitalproject.org/t/urban-stormwater-retention-model-indexerror/3052/2) was encountering an interesting bug:

```python
2022-09-15 13:30:43,132 (pygeoprocessing.geoprocessing) geoprocessing.raster_calculator(498) ERROR exception encountered in raster_calculator
Traceback (most recent call last):
  File "pygeoprocessing\geoprocessing.py", line 443, in raster_calculator
  File "natcap\invest\stormwater.py", line 671, in ratio_op
IndexError: arrays used as indices must be of integer (or boolean) type
2022-09-15 13:30:43,133 (pygeoprocessing.geoprocessing) geoprocessing.raster_calculator(513) INFO Waiting for raster stats worker result.
2022-09-15 13:30:43,134 (pygeoprocessing.geoprocessing_core) threading.run(946) WARNING No valid pixels were received, sending None.
2022-09-15 13:30:43,134 (taskgraph.Task) Task.add_task(706) ERROR Something went wrong when adding task calculate stormwater retention ratio (1), terminating taskgraph.
Traceback (most recent call last):
  File "taskgraph\Task.py", line 674, in add_task
  File "taskgraph\Task.py", line 1093, in _call
  File "natcap\invest\stormwater.py", line 674, in lookup_ratios
  File "pygeoprocessing\geoprocessing.py", line 443, in raster_calculator
  File "natcap\invest\stormwater.py", line 671, in ratio_op
IndexError: arrays used as indices must be of integer (or boolean) type
```

This occurs in: https://github.com/natcap/invest/blob/41070ee205506d8000f14ba2d81a53567d1882b7/src/natcap/invest/stormwater.py#L658 

During a numpy array indexing call the soil array is being used as an index, but it has a floating point types, which causes the error.

The solution here could be to just handle floating point soil inputs or to validate upfront that the soil raster is of integer type.

### PR DESCRIPTION
**Pull request by emlys**
## Description
Fixes #1084 

## Checklist
- [x] Updated HISTORY.rst and link to any relevant issue (if these changes are user-facing)
- [ ] ~Updated the user's guide (if needed)~
- [ ] ~Tested the affected models' UIs (if relevant)~

### CODE REVIEW

File: tests/test_stormwater.py [hunk 1]
```diff
@@ -798,6 +798,34 @@ def test_validate(self):
...
 
+    def test_validate_noninteger_soil_raster(self):
+        """Stormwater: test arg validation."""
+        from natcap.invest import stormwater, validation
+
+        soil_array = numpy.array([[1, 2], [3, 4]], dtype=numpy.float32)
+        soil_path = os.path.join(self.workspace_dir, 'soils.tif')
+        to_raster(soil_array, soil_path, pixel_size=(10, -10))
+
+        # test args missing necessary values for adjust ratios
```
Comment (davemfish) [inline tests/test_stormwater.py:809]: This comment seems out of place. I wouldn't want us to be confused in the future if this is not actually what's being tested here.

File: HISTORY.rst [hunk 2]
```diff
@@ -64,6 +64,9 @@ Unreleased Changes
       flag, leading to a perceived buffering of spatial criteria in certain
       cases.  In InVEST 3.9.0, these were rasterized with ``ALL_TOUCHED=FALSE``.
       https://github.com/natcap/invest/issues/1120
+* Stormwater
```
Comment (davemfish) [inline HISTORY.rst:67]: Technically we list this model as "Urban Stormwater Retention" in the key at the top of this file.

**Review by davemfish — CHANGES_REQUESTED** [review]
Thanks, @emlys , this looks like it does the job nicely!

I left a couple really minor comments that will make our lives very marginally easier in the future, if you don't mind.

File: tests/test_stormwater.py (hunk 1 above)
Comment (emlys) [reply tests/test_stormwater.py:809]: Good catch, I accidentally pasted that in from a different test.

**Review by davemfish — APPROVED** [review]
Thanks! Looks like the two failing checks are MacOS timeout issues, unrelated to these changes.

**Pull request merged**